src/compiler/build/validate-files.ts

```diff
@@ -2,7 +2,19 @@ import type * as d from '../../declarations';
 import { validateBuildPackageJson } from '../types/validate-build-package-json';
 import { validateManifestJson } from '../html/validate-manifest-json';
 
-export const validateBuildFiles = (config: d.Config, compilerCtx: d.CompilerCtx, buildCtx: d.BuildCtx) => {
+/**
+ * Validate the existence and contents of certain files that were generated after writing the results of the build to
+ * disk
+ * @param config the Stencil configuration used for the build
+ * @param compilerCtx the compiler context associated with the build
+ * @param buildCtx the build context associated with the current build
+ * @returns an array containing empty-Promise results
+ */
+export const validateBuildFiles = (
+  config: d.ValidatedConfig,
+  compilerCtx: d.CompilerCtx,
+  buildCtx: d.BuildCtx
+): Promise<(void | void[])[]> => {
```

Comment:
Didn't know `Promise<(void | void[])[]>` and `Promise<[void, void[]]>` were the same type

Reply:
The `[]` distributes over the parens, so `Promise<(void | void[])[]>` is equivalent to `Promise<void[] | void[][]>`, rather than a tuple consisting of something assignable to `void` in the first position and something assignable to `void[]` in the second.

Surprisingly, TS lets you do something like: 😆 
```ts
type Voidey = [void, void[]];
const a: Voidey = [undefined, [undefined, undefined, undefined]];
```

Reply:
Yeah I knew the `[]` distributed, but I woulda thought TS treated `void[] | void[][]` and `[void, void[]]` as different types 🤷‍♂️

Reply:
Hmmm I think we've got some disconnect here. They are different types - `[void, void[]]` is a tuple with two fields, whereas `void[] | void[][]` is either a `void` array or a `void` array of arrays (without a size bound).